### ISSUE TITLE
Can I add an ID to a paragraph? Convenient data location

### ISSUE THREAD

**Issue by verydemo** [ONLYOFFICE/DocumentServer#1362]
Can I add an ID to a paragraph? Convenient data location

**Comment by ShockwaveNN** [comment]
Could you describe your request in more details

Why do you need this?

As far as I aware MS Word do not allow changing paragraph ID either

**Comment by verydemo** [comment]
![image](https://user-images.githubusercontent.com/29354679/127470441-62ccafd7-d3e8-431f-b137-ec2be29015de.png)

MS hava a ParaID , I don't need to change it. I want to locate paragraph  .

**Comment by ShockwaveNN** [comment]
Hm, now I understand you

Something like

Add ability to get parargaph by id, not by index to [`GetParagraph`](https://api.onlyoffice.com/docbuilder/textdocumentapi/apirange/getparagraph) method?

**Comment by verydemo** [comment]
> Hm, now I understand you
> 
> Something like
> 
> Add ability to get parargaph by id, not by index to [`GetParagraph`](https://api.onlyoffice.com/docbuilder/textdocumentapi/apirange/getparagraph) method?
 The index not unique in Third-party programs and index  is dynamic.

**Comment by verydemo** [comment]
Current Solutions:
```
for (var i = 0; i < arrParagraphs.length; i++) {
    var oParagraph = arrParagraphs[i];
    if (oParagraph.GetClassType() == "paragraph") {
        if(!oParagraph.hasOwnProperty("InsertInContentControl")){continue;}
        oBlockLvlSdt = oParagraph.InsertInContentControl(1);
        oBlockLvlSdt.SetLock(3);
        oBlockLvlSdt.SetTag(uuid);
        console.log("i value: ",i)
    }
}
```

I add ContentControl to each paragraph, each ContentControl set tag for uuid. both in the third-party app to find the paragraph, but also in onlyoffice to find the corresponding paragraph.

**Comment by ShockwaveNN** [comment]
So in result we need to methods

1. Method which will add id to paragraph
2. Method which will get paragraph by id

**Comment by verydemo** [comment]
> * Method which will add id to paragraph
> * Method which will get paragraph by id

Yes, if there are these two methods, OO will better interface with third-party programs

**Comment by ShockwaveNN** [comment]
Thanks for your idea, I've created issue 51600 in our private issue tracker to discuss it with our dev team
